### PR TITLE
Support for Distribution path mapping with AQL

### PR DESCRIPTION
**Pull request by YardenEdery**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Added support of "mappings" property to file spec in order to have the ability to control where to distribute the artifacts that are being added dynamically when using AQL in the release bundle that is being created.

When looking for docker artifacts using AQL, the release bundle also includes, in the distribution server backend, all the layers for a manifest.json and all of the manifest.json files and their layers if its a list.manifest.json

From now on the "mappings" will take affect on those none specified added manifests and layers and will allow the users a native way of creating the file spec for distribution commands.

FYI This couldn't be achieved using "target" for the list.manifest.json example.

### CODE REVIEW

**Comment by YardenEdery** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Yarden-Edery** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Yarden-Edery** [comment]
recheck

**Comment by eyalbe4** [comment]
Thanks for creating this PR @YardenEdery!
Can you please add full description for this PR?

File: tests/distribution_test.go [hunk 1]
```diff
@@ -355,6 +364,52 @@ func createSignSyncDistributeDelete(t *testing.T) {
...
+
+	// Make sure <RtTargetRepo>/b.out does exist in Artifactory
+	searchParams := artifactoryServices.NewSearchParams()
+	searchParams.Pattern = getRtTargetRepo() + "b.out"
+	reader, err := testsSearchService.Search(searchParams)
+	assert.NoError(t, err)
+	readerCloseAndAssert(t, reader)
+	length, err := reader.Length()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, length)
```
Comment (eyalbe4) [inline tests/distribution_test.go:407]: The above code lines seem to be duplicated multiple times in this file. Let's move into a function and reuse the code.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: tests/distribution_test.go [hunk 2]
```diff
@@ -355,6 +364,52 @@ func createSignSyncDistributeDelete(t *testing.T) {
 func createDistributeMapping(t *testing.T) {
 	bundleName := initRemoteDistributionTest(t, "client-test-bundle-"+getRunId())
 
+	// Create release bundle with path mapping from <RtTargetRepo>/b.in to <RtTargetRepo>/b.out
+	createBundleParams := services.NewCreateReleaseBundleParams(bundleName, bundleVersion)
+	createBundleParams.SpecFiles = []*utils.CommonParams{
+		{
+			Aql: utils.Aql{
+				ItemsFind: "{\"$or\":[{\"$and\":[{\"repo\":{\"$match\": \"" + strings.TrimSuffix(getRtTargetRepo(), "/") + "\"},\"name\":{\"$match\":\"b.in\"}}]}]}",
```
Comment (yahavi) [inline tests/distribution_test.go:372]:
```suggestion
				ItemsFind: "{\"$or\":[{\"$and\":[{\"repo\":{\"$match\":\"" + strings.TrimSuffix(getRtTargetRepo(), "/") + "\"},\"name\":{\"$match\":\"b.in\"}}]}]}",
```

File: tests/distribution_test.go [hunk 3]
```diff
@@ -388,7 +443,54 @@ func createDistributeMapping(t *testing.T) {
...
-func createDistributeMappingPlaceholder(t *testing.T) {
+func createDistributeMappingWithPlaceholder(t *testing.T) {
+	bundleName := initRemoteDistributionTest(t, "client-test-bundle-"+getRunId())
+
+	// Create release bundle with path mapping from <RtTargetRepo>/b.in to <RtTargetRepo>/b.out
+	createBundleParams := services.NewCreateReleaseBundleParams(bundleName, bundleVersion)
+	createBundleParams.SpecFiles = []*utils.CommonParams{
+		{
+			Aql: utils.Aql{
+				ItemsFind: "{\"$or\":[{\"$and\":[{\"repo\":{\"$match\": \"" + strings.TrimSuffix(getRtTargetRepo(), "/") + "\"},\"name\":{\"$match\":\"*.in\"}}]}]}",
```
Comment (yahavi) [inline tests/distribution_test.go:454]:
```suggestion
				ItemsFind: "{\"$or\":[{\"$and\":[{\"repo\":{\"$match\":\"" + strings.TrimSuffix(getRtTargetRepo(), "/") + "\"},\"name\":{\"$match\":\"*.in\"}}]}]}",
```

File: tests/distribution_test.go [hunk 4]
```diff
@@ -388,7 +443,54 @@ func createDistributeMapping(t *testing.T) {
...
+			PathMapping: utils.PathMapping{
+				Input:  "(" + getRtTargetRepo() + ")" + "(.*).in",
+				Output: "$1$2.out",
+			},
+		},
+	}
+
+	createBundleParams.SignImmediately = true
+	summary, err := testsBundleCreateService.CreateReleaseBundle(createBundleParams)
+	if !assert.NoError(t, err) {
```
Comment (yahavi) [inline tests/distribution_test.go:465]: For better readability and shorter code, let's not use "ifs" in tests. The "assert" and/or "require" alone should help us to eliminate them.

**Review by yahavi — CHANGES_REQUESTED** [review]

**Review by yahavi — APPROVED** [review]
Well done @Yarden-Edery!
Thanks.

**Pull request merged**